srml/support/src/dispatch.rs

```diff
@@ -206,15 +206,21 @@ macro_rules! decl_module {
 	// Entry point #1.
 	(
 		$(#[$attr:meta])*
-		pub struct $mod_type:ident<$trait_instance:ident: $trait_name:ident$(<I>, I: $instantiable:path $(= $module_default_instance:path)?)?>
-		for enum $call_type:ident where origin: $origin_type:ty {
+		pub struct $mod_type:ident<
+			$trait_instance:ident: $trait_name:ident
+			$( <I>, I: $instantiable:path $( = $module_default_instance:path )? )?
+		>
+		for enum $call_type:ident where origin: $origin_type:ty $(, $other_where_bounds:ty: $other_stuff:path )* {
```

Comment:
oh ok I see only `type: trait` is allowed here: no lifetime and no +

Reply:
Lifetime should not required and `+` is not doable in declarative macros :(